prow/rewriteConfig.go

```diff
@@ -0,0 +1,194 @@
+			checkForMaster := line[spacesForBranch:]
+			if strings.Contains(checkForMaster, "master:") {
+				masterStart = j + 1
+			}
+		}
+
+		if masterStart != -1 && masterStop == -1 && j > masterStart {
+			if countLeadingSpace(line) <= spacesForBranch {
+				masterStop = j - 1
+			}
+		}
+	}
+
+	if masterStart != -1 && masterStop == -1 {
+		masterStop = len(branchLines) - 1
+	}
+
+	if masterStart != -1 && masterStop != -1 {
+
+		contextStart := -1
+		contextStop := -1
+		contextStrings := -1
+
+		secondBranchSpaces := countLeadingSpace(branchLines[masterStart])
+
+		for m := 0; m < masterStop-masterStart+1; m++ {
+			masterLine := branchLines[m+masterStart]
+			if strings.Contains(masterLine, "contexts:") {
+				contextStart = m + 1
+				contextStrings = countLeadingSpace(masterLine)
+			}
+
+			if contextStart != -1 && contextStrings != -1 && contextStop == -1 && m > contextStart {
+
+				if strings.Compare(string(masterLine[contextStrings]), "-") != 0 {
+					contextStop = m
+				}
+			}
+		}
+
+		if contextStart != -1 && contextStop == -1 {
+			contextStop = masterStop - masterStart + 1
+		}
+		containsNeedsAdmin := false
+		newBranchSlice := copySlice(branchLines[masterStart : masterStop+1])
+		if contextStart != -1 && contextStop != -1 {
+			for n := contextStart; n < contextStop; n++ {
+				contextLine := branchLines[n+masterStart]
+
+				if strings.Contains(contextLine, "merges-blocked-needs-admin") {
+					containsNeedsAdmin = true
+					break
+				}
+			}
+
+			if !containsNeedsAdmin {
+				numSpaces := countLeadingSpace(newBranchSlice[contextStart])
+				stringAdmin := strings.Repeat(" ", numSpaces) + "- \"merges-blocked-needs-admin\""
+
+				newBranchSlice = appendAtIndex(newBranchSlice, stringAdmin, contextStop-1)
+			}
+
+		} else {
+
+			if len(newBranchSlice[len(newBranchSlice)-1]) == countLeadingSpace(newBranchSlice[len(newBranchSlice)-1]) {
+				newBranchSlice = newBranchSlice[:len(newBranchSlice)-1]
+			}
+			statusCheckString := strings.Repeat(" ", secondBranchSpaces) + "required_status_checks:"
+			contextsString := strings.Repeat(" ", secondBranchSpaces+2) + "contexts:"
+			needsAdminString := strings.Repeat(" ", secondBranchSpaces+2) + "- \"merges-blocked-needs-admin\""
+			newBranchSlice = append(newBranchSlice, statusCheckString, contextsString, needsAdminString)
+		}
+
+		newBranchLine := strings.Repeat(" ", spacesForBranch) + newBranch + ":"
+		newBranchSlice = appendAtIndex(newBranchSlice, newBranchLine, 0)
+		masterBranch := strings.Join(newBranchSlice, "\n")
+		return masterStart - 1, masterBranch
+	}
+	return -1, ""
+}
+
+// Convert read result into string and split it based on "branches:" to make it
+// simpler to find "master:" and its contents. Get output from getContentOfNewBranch().
+// Add new branch content before master section and rejoin the string.
+func findMaster(source []byte, newBranch string) string {
+	sourceString := string(source)
+	eachBranches := strings.Split(sourceString, "branches:")
```

Comment:
Instead of looking for all master branches, can you add the check for repos, we want to add new branch info only for 
[proxy, istio, istio-releases]